src/main/java/hudson/tasks/junit/TestResult.java

```diff
@@ -306,14 +310,20 @@ public void parse(File reportFile) throws IOException {
         } catch (RuntimeException e) {
             throw new IOException("Failed to read "+reportFile,e);
         } catch (DocumentException e) {
+            String reportFileContent = null;
+            try {
+                reportFileContent = Files.toString(reportFile, UTF_8);
+            } catch (Exception ee) {
+                /* No need to further explain the error if we end up here. */
+            }
             if (!reportFile.getPath().endsWith(".xml")) {
-                throw new IOException("Failed to read "+reportFile+"\n"+
-                    "Is this really a JUnit report file? Your configuration must be matching too many files",e);
+                throw new IOException("Failed to parse "+reportFile+"\n"+
+                    "Is this really a JUnit report file? Your configuration may be matching too many files? The file contains:\n\n" + reportFileContent + "\n\n",e);
```

Comment:
Full contents shouldn't be transferred. Could be gigabytes of data. Perhaps the first 100 bytes, the last 100 bytes, if ASCII readable with other chars as hex codes (I think the remoting stream diagnosis looks like that and it helps with binary crap), and the total length?